### ISSUE TITLE
Self-import is not flagged as unused

### ISSUE THREAD

**Issue by TWiStErRob** [pinterest/ktlint#2855]
We moved some classes around as a refactor and we ended up accidentally importing a class from the same package:

## Steps to Reproduce
Production code (src/main/kotlin/foo/bar)
```kotlin
package foo.bar

class Production
```

Test code (src/test/kotlin/foo/bar)
```kotlin
package foo.bar

import foo.bar.Production

class ProductionTest
```

## Expected Behavior

(`gradlew spotlessApply`)

```patch
package foo.bar

-import foo.bar.Production

class ProductionTest
```

Note: Android Studio shows it as unused.

## Observed Behavior

No change.

## Your Environment

* Version of ktlint used: 1.4.0
* Relevant parts of the `.editorconfig` settings: `ktlint_code_style = android_studio`
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): Spotless 7.0.0.BETA4
* Version of Gradle used (if applicable): 8.10.2
* Operating System and version: Mac 15 (local) and Ubuntu 24 (CI)

**Comment by paul-dingemans** [comment]
Previously imports like this were removed. But in some cases this would lead to code which no longer could be compile (see #2821). Therefore ktlint is rather conservative with removing the import. It is better to have an unused import compared to having code that cannot be compiled.

**Comment by paul-dingemans** [comment]
Ktlint should take all lint/format decisions based on the file it scans only. It can not take into account any information which is to be retrieved from any other file.
